src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ExceptionsStoreTests.cs

```diff
@@ -297,23 +307,36 @@ private sealed class ThresholdCallback : ExceptionsStoreCallbackBase
             private int _addCount;
             private int _removeCount;
 
-            public ThresholdCallback(int addThreshold, int removeThreshold = 0)
+            public ThresholdCallback(ITestOutputHelper outputHelper, int addThreshold, int removeThreshold = 0)
             {
                 _addThreshold = addThreshold;
                 _removeThreshold = removeThreshold;
+                _outputHelper = outputHelper;
 
                 SetIfAtTarget(_addCompletionSource, 0, addThreshold);
                 SetIfAtTarget(_removeCompletionSource, 0, removeThreshold);
+
+                _outputHelper.WriteLine($"{nameof(ThresholdCallback)}.Ctor(Add={addThreshold}, Remove={removeThreshold})");
             }
 
             public override void AfterAdd(IExceptionInstance instance)
             {
-                SetIfAtTarget(_addCompletionSource, ++_addCount, _addThreshold);
+                _outputHelper.WriteLine($"{nameof(ThresholdCallback)}.{nameof(AfterAdd)}");
+
+                if (SetIfAtTarget(_addCompletionSource, ++_addCount, _addThreshold))
+                {
+                    _outputHelper.WriteLine($"{nameof(ThresholdCallback)}.{nameof(AfterAdd)} threshold reached");
+                }
             }
 
             public override void AfterRemove(IExceptionInstance instance)
             {
-                SetIfAtTarget(_removeCompletionSource, ++_removeCount, _removeThreshold);
+                _outputHelper.WriteLine($"{nameof(ThresholdCallback)}.{nameof(AfterRemove)}");
```

Comment:
Could [CallerMemberName] simplify some of these log entries

Reply:
I don't think so. I'm not trying to capture the caller, I want to log that the current method is executing.